### PR TITLE
Gx cooldown 4   deprecations and be api update

### PR DESCRIPTION
**Pull request by EbrithilNogare**
#### Changelog notes 

AdultCount and ChildCount were marked as deprecated.

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to the correct place in the JSON.
- [ ] New properties in the table are added to the correct place.
- [ ] Correct formatting:
  - [ ] Spacing is consistent between titles, sections, tables, ...
  - [ ] Correct JSON format - indentation.
- [ ] DateTime properties should always be defined in ISO format.

### CODE REVIEW

File: booking-engine-api/operations/hotels.md
Comment (zdrazil) [inline booking-engine-api/operations/hotels.md:426]: @progresak Should we mention that one of them needs to be adult? Do you know? I think we had a limitation like that?

File: booking-engine-api/operations/hotels.md
Comment (zdrazil) [inline booking-engine-api/operations/hotels.md:437]: @progresak @EbrithilNogare If we update the docs like this, I'm not sure people will be able to figure out, where they should find the ageCategoryId.

Additionally, it might be hard for them to figure out which numbers are meant for children and which are for adults, especially for those who need to migrate from the old API.

I don't know if you already discussed this together?

**Comment by zdrazil** [comment]
@progresak I'm not sure what the process is for these doc updates? Who needs to approve?

File: booking-engine-api/operations/hotels.md
Comment (EbrithilNogare) [reply booking-engine-api/operations/hotels.md]: Good point, I'm having plan to add pretier configs (here and to globalization) in one of the next cooldowns.

File: booking-engine-api/deprecations/README.md
Comment (EbrithilNogare) [reply booking-engine-api/deprecations/README.md:31]: I was not sure when the deprecation happened and not to look confusing I filled date that should be soon after release of this PR (I hoped I do it faster).

File: booking-engine-api/operations/hotels.md
Comment (EbrithilNogare) [reply booking-engine-api/operations/hotels.md:426]: API has valid response when at least one OccupancyData is present, does not depend on adult/child.

File: booking-engine-api/operations/hotels.md
Comment (zdrazil) [reply booking-engine-api/operations/hotels.md]: Ha, cool 👍 So I won't do it 😀

File: booking-engine-api/operations/hotels.md
Comment (zdrazil) [reply booking-engine-api/operations/hotels.md:426]: But for the request?

File: booking-engine-api/operations/hotels.md
Comment (EbrithilNogare) [reply booking-engine-api/operations/hotels.md:437]: That is a pretty good question as we don't have it on Mews help. I have two possible solutions, which do you prefer?
- We can add a mention to [mews help ](https://help.mews.com/s/article/Configuring-age-categories-in-Mews?language=en_US)
- We can it here to look like this:

| `AgeCategoryId` | string | required | Age category identifier (found in url in commander: `/Commander/.../AgeCategory/Detail/{AgeCategoryId})`. |
| :-- | :-- | :-- | :-- |

File: booking-engine-api/deprecations/README.md
Comment (progresak) [reply booking-engine-api/deprecations/README.md:31]: I believe it is fine to put there this date.

File: booking-engine-api/operations/hotels.md
Comment (progresak) [reply booking-engine-api/operations/hotels.md:437]: I'd go with the proposed solution from Slack:
- Add it here into the documentation. It is the fastest way.
same way like: https://mews-systems.gitbook.io/booking-engine-guide/faq#where-can-i-get-configuration-id

**Comment by MikeAdamsMews** [comment]
> @progresak I'm not sure what the process is for these doc updates? Who needs to approve?

Me :-) There is a repository rule set up for Connector API docs that means every change on those docs needs to be approved by someone from the tech team (meaning the developers) and someone from Product Content (which basically means me at the moment). It would be good if the same system was in place for Booking Engine Guide. Is that something you could set up @progresak?

**Comment by progresak** [comment]
@MikeAdamsMews `Booking Engine API` section from this repository requires `tech.fe.connectivity.owners` review because on Sep 22, 2023 we handed over the responsibility and **code**-ownership of public API to Connectivity team. 

Mike, I will set you up as code-owner to the rest of the files from this repo, as you are the only Tech writer I know - that means
- When we touch `Booking Engine API` section - it will assign `tech.fe.connectivity.owners`
- When we touch everything else, it will assign only you - connectivity doesn't care about the rest docs not related to API.

Now you can do the review then.

**Comment by EbrithilNogare** [comment]
Added age category id "how to find" into faq.

**Review by progresak — APPROVED** [review]

**Comment by MikeAdamsMews** [comment]
Sorry for the delay in approving this, after my previous statement! I've been on Connectivity events all this week and I'm only now able to look at it properly.

File: FAQ/README.md
Comment (MikeAdamsMews) [inline FAQ/README.md:50]:
```suggestion
## Where can I get Age Category ID?
```
Just for consistency with `Configuration ID` and `City ID`.

File: FAQ/README.md
Comment (MikeAdamsMews) [inline FAQ/README.md:59]:
```suggestion
* In the URL there is an ID at the end, right after `/Detail/`. This is the age category ID. It has the format `aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee`.
```

File: FAQ/README.md
Comment (MikeAdamsMews) [inline FAQ/README.md:60]: It's a small point, but I would remove the word DEVELOPMENT from the image (where the screnshot is taken from the Development Server), what do you think?

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:428]:
```suggestion
| ~~`ChildCount`~~ | ~~number~~ | ~~optional~~ | ~~Requested number of children.~~ **Deprecated!** |
```

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:427]:
```suggestion
| ~~`AdultCount`~~ | ~~number~~ | ~~optional~~ | ~~Requested number of adults. If provided together with `ChildCount`, then `RoomOccupancyAvailabilities` will be computed only for that combination instead of all possible. If `RoomCategory` doesn’t support the given values, then the nearest applicable values are found.~~ **Deprecated!** |
```

File: booking-engine-api/deprecations/README.md
Comment (MikeAdamsMews) [reply booking-engine-api/deprecations/README.md:31]: Regarding the dates, what we do in the Connectivity team is just put in the correct date at the time of writing (and same for the changelog date) and then when the PR is ready to be merged, at that point the person doing the merge should update them accordingly.

**Review by MikeAdamsMews — COMMENTED** [review]
I've not finished my review yet, so far I have a few minor suggestions plus a couple of additional points:

1. Please add a changelog entry, following the style of other entries, e.g. beginning "API:" if it's a change in the API part of the Guide.
2. This one should probably be on a separate PR as it's nothing to do with this PR, but I noticed that the Deprecations are listed in the opposite date order to how they are listed in the Connector API Docs, i.e. oldest first instead of newest first.

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:438]:
```suggestion
| `PersonCount` | number | required | Number of people in the given age category. |
```

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:437]:
```suggestion
| `AgeCategoryId` | string | required | Age category identifier - see [Where can I get Age Category ID?](../../FAQ/README.md#where-can-i-get-age-category-id). |
```

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:579]:
```suggestion
| ~~`AdultCount`~~ | ~~number~~ | ~~required~~ | ~~Number of adults for the associated pricing.~~ **Deprecated!** |
```

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/hotels.md:580]:
```suggestion
| ~~`ChildCount`~~ | ~~number~~ | ~~required~~ | ~~Number of childs for the associated pricing.~~ **Deprecated!** |
```

File: booking-engine-api/operations/reservations.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/reservations.md:53]:
```suggestion
| ~~`AdultCount`~~ | ~~number~~ | ~~required~~ | ~~Number of adults.~~ **Deprecated!** |
```

File: booking-engine-api/operations/reservations.md
Comment (MikeAdamsMews) [inline booking-engine-api/operations/reservations.md:54]:
```suggestion
| ~~`ChildCount`~~ | ~~number~~ | ~~required~~ | ~~Number of children.~~ **Deprecated!** |
```

File: booking-engine-api/operations/hotels.md
Comment (MikeAdamsMews) [reply booking-engine-api/operations/hotels.md:437]: Ah, I just realised that Occupancy Data is already defined in reservations.md. In that case, rather than duplicate it, I think it would be better to reference it from hotels.md, i.e. ```[Occupancy Data](reservations.md#occupancy-data)```. At the same time, the definition in reservations.md should be updated to reflect the changes suggested here, as follows:

#### Occupancy Data

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `AgeCategoryId` | string | required | Age category identifier - see [Where can I get Age Category ID?](../../FAQ/README.md#where-can-i-get-age-category-id). |
| `PersonCount` | number | required | Number of people in the given age category. |

I hope that's clear :-)

**Review by MikeAdamsMews — DISMISSED** [review]

File: booking-engine-api/deprecations/README.md
Comment (MikeAdamsMews) [inline booking-engine-api/deprecations/README.md:31]: The **deprecated** date should be the same date as the Changelog, i.e. now. This is the date when we announce the deprecation. The **discontinued** date is the date at which the feature will be permanently deleted or discontinued. See [Deprecation policy](https://mews-systems.gitbook.io/open-api/staying-up-to-date/deprecations-policy).

Comment (MikeAdamsMews) [inline booking-engine-api/deprecations/README.md:31]: The deprecation applies to more than just **Get reservation price**, we should list all the affected operations. It's a bit of a chore, but the purpose of the table is to make it easy for users to know if a particular deprecation affects them or not.

**Review by MikeAdamsMews — COMMENTED** [review]
There are a few things that didn't get addressed from my first review. We also need a Changelog entry, for example as follows:

## 24th November 2023
* API: `AdultCount` and `ChildCount` deprecated and replaced by `OccupancyData` in [Get availability](../booking-engine-api/operations/hotels.md#get-availability), [Get reservations pricing](../booking-engine-api/operations/reservations.md#get-reservations-pricing) and [Get reservation price](../booking-engine-api/operations/reservations.md#get-reservation-price)
* General: Added [Where can I get Age Category ID?](../FAQ/README.md#where-can-i-get-age-category-id)

**Pull request merged**
